specs/interop/predeploys.md

```diff
@@ -0,0 +1,260 @@
+
+The `CrossL2Inbox` is responsible for executing a cross chain message on the destination chain.
+It is permissionless to execute a cross chain message on behalf of any user.
+
+To ensure safety of the protocol, the [Message Invariants](./messaging.md#messaging-invariants) must be enforced.
+
+### Message execution arguments
+
+The following fields are required for executing a cross chain message:
+
+[message payload]: ./messaging.md#message-payload
+[`Identifier`]: ./messaging.md#message-identifier
+
+| Name      | Type         | Description                                             |
+|-----------|--------------|---------------------------------------------------------|
+| `_msg`    | `bytes`      | The [message payload], matching the initiating message. |
+| `_id`     | `Identifier` | A [`Identifier`] pointing to the initiating message.    |
+| `_target` | `address`    | Account that is called with `_msg`.                     |
+
+#### `_msg`
+
+The [message payload] of the executing message.
+
+This must match the emitted payload of the initiating message identified by `_id`.
+
+#### `_id`
+
+A pointer to the `_msg` in a remote (or local) chain.
+
+The message [`Identifier`] of the executing message.
+This is required to enforce the message executes an existing and valid initiating message.
+
+By including the [`Identifier`] in the calldata, it makes static analysis much easier for block builders.
+It is impossible to check that the [`Identifier`] matches the cross chain message on chain. If the block
+builder includes a message that does not correspond to the [`Identifier`]`, their block will be reorganized
+by the derivation pipeline.
+
+A possible upgrade path to this contract would involve adding a new function. If any fields in the [`Identifier`]
+change, then a new 4byte selector will be generated by solc.
+
+#### `_target`
+
+Messages are broadcast, not directed. Upon execution the caller can specify which `address` to target:
+there is no protocol enforcement on what this value is.
+
+The `_target` is called with the `_msg` as input.
+In practice, the `_target` will be a contract that needs to know the schema of the `_msg` so that it can be decoded.
+It MAY call back to the `CrossL2Inbox` to authenticate
+properties about the `_msg` using the information in the `Identifier`.
+
+### Reference implementation
+
+A simple implementation of the `executeMessage` function is included below.
+
+```solidity
+function executeMessage(address _target, bytes calldata _msg, Identifier calldata _id) public payable {
+    require(msg.sender == tx.origin);
+    require(_id.timestamp <= block.timestamp);
+    require(L1Block.isInDependencySet(_id.chainid));
+
+    assembly {
+      tstore(ORIGIN_SLOT, _id.origin)
+      tstore(BLOCKNUMBER_SLOT, _id.blocknumber)
+      tstore(LOG_INDEX_SLOT, _id.logIndex)
+      tstore(TIMESTAMP_SLOT, _id.timestamp)
+      tstore(CHAINID_SLOT, _id.chainid)
+    }
```

Comment:
What ends up reading these slots? Is this info made available for `_target` to query if needed since `_id` is `calldata` and wouldn't otherwise be available?

Reply:
> The Identifier MUST be exposed via public getters so that contracts can call back to authenticate properties about the _msg.

Yep that's it